src/librustc_mir/interpret/machine.rs

```diff
@@ -192,6 +192,8 @@ pub trait Machine<'mir, 'tcx>: Sized {
     ) -> InterpResult<'tcx>;
 
     /// Called to read the specified `local` from the `frame`.
+    /// Since reading a ZST is not actually accessing memory or locals, this is never invoked
+    /// for ZST reads.
```

Comment:
To be clear, this was already the case before but now is documented better?

Reply:
yes